src/models.py

```diff
@@ -931,6 +957,42 @@ def _lm_forward(self, batch, task, predict):
             pass
         return out
 
+    def _lm_only_forward(self, batch, task, predict):
+        """Only forward pass for LM model - consider for non-bidirectional models
```

Comment:
Does forward mean feedforward (vs backprop) or left-to-right (vs right-to-left)? Clarify and rename.

Reply:
If any of this code comes from Shen, say so, and include a link.

Reply:
Renamed to clarify left-to-right movement and added some brief comments. This code isn't from Shen. On a side note: Gave credits to Shen's github in the README file. I don't know if that's the right place?